### PR TITLE
Add Allowed Registries

### LINKED ISSUE #3518: AllowedRegistry UI
Design and implement UI for AllowedRegistry

### PR DESCRIPTION
**Pull request by kgroschoff**
### What this PR does / why we need it
Adds Allowed Registries to UI. Note that this is **only** an ee feature.

### Which issue(s) this PR fixes

Fixes #3518

### Special notes for your reviewer

allowed1
allowed2

### Release note

```release-note
EE: Added Allowed Registries
```

### CODE REVIEW

**Comment by kgroschoff** [comment]
/test pre-dashboard-test-e2e

@maciaszczykm / @floreks PTAL

File: src/app/settings/admin/opa/component.ts [hunk 1]
```diff
@@ -11,11 +11,16 @@
...
+import {getEditionDirName} from '@app/dynamic/common';
 
 @Component({
   selector: 'km-admin-settings-opa',
   templateUrl: './template.html',
 })
 export class AdminSettingsOPAComponent {
   readonly Context = Context;
+
+  isEE(): boolean {
```
Comment (maciaszczykm) [inline src/app/settings/admin/opa/component.ts:23]: Can we make this method part of the `@app/dynamic/common` and use it in all other places?

Comment (maciaszczykm) [reply src/app/settings/admin/opa/component.ts:23]: Also please change to `isEnterpriseEdition()`.

**Comment by maciaszczykm** [comment]
I've one minor comment. Other than that looks good.

File: src/app/settings/admin/opa/component.ts (hunk 1 above)
Comment (kgroschoff) [reply src/app/settings/admin/opa/component.ts:23]: done :)

File: src/app/settings/admin/opa/template.html [hunk 2]
```diff
@@ -19,4 +19,9 @@
   <km-tab label="Default Constraints">
     <km-default-constraint-list></km-default-constraint-list>
   </km-tab>
+  <km-tab label="Allowed Registries"
+          *ngIf="isEE()">
```
Comment (floreks) [inline src/app/settings/admin/opa/template.html]: The idea of using dynamic ce/ee modules was to not use this kind of logic and let the router load proper ce/ee version instead. It should be hidden inside and the only thing here should be the `router-outlet`.

Comment (kgroschoff) [reply src/app/settings/admin/opa/template.html]: Yes, agree, that was my first thought too, but I did not manage to load this tab label from the router-outlet for some reason.

Comment (maciaszczykm) [reply src/app/settings/admin/opa/template.html]: I think there might be a problem with `km-tab-card` not recognizing the `km-tab` if it comes from the router outlet. I would check that first.

Comment (kgroschoff) [reply src/app/settings/admin/opa/template.html]: Something like this, yeah. The tab isn't in the `inputTabs` list in `km-tab-card`, even though it seems that `km-tab` is "receiving" it. Therefor it isn't rendered.
Investigated a bit more and tried out a lot, but still didn't manage to create a working solution. 🤷 

Is someone having an idea and wants to take a look on this problem?

Comment (floreks) [reply src/app/settings/admin/opa/template.html]: I'll take a look

Comment (floreks) [reply src/app/settings/admin/opa/template.html]: Added dynamic tab support. Should be ok now.

Comment (kgroschoff) [reply src/app/settings/admin/opa/template.html]: Thanks, looks good. 👍 
I spot a lot of parts, that I've also tried, but obviously not in the right combination.

File: src/app/settings/admin/opa/component.ts [hunk 3]
```diff
@@ -9,18 +9,28 @@
...
+  private _dynamicTabs = new Set<DynamicTabComponent>();
 
-  isEnterpriseEdition(): boolean {
-    return isEnterpriseEdition();
+  get dynamicTabs(): DynamicTabComponent[] {
+    return [...this._dynamicTabs];
+  }
+
+  onActivate(component: DynamicTab): void {
+    component.onTabReady.pipe(takeUntil(this._unsubscribe)).subscribe(tab => this._dynamicTabs.add(tab));
```
Comment (floreks) [inline src/app/settings/admin/opa/component.ts:34]: I have forgotten to add `ngOnDestroy` and unsubscribe part here. Please do.

Comment (kgroschoff) [reply src/app/settings/admin/opa/component.ts:34]: 👍

**Comment by kgroschoff** [comment]
/retest

@maciaszczykm / @floreks PTAL

**Comment by kgroschoff** [comment]
PTAL @maciaszczykm / @floreks 
Adjusted the code according to latest changes on how to load dynamic modules.

**Comment by maciaszczykm** [comment]
@kgroschoff Thanks a lot!

/lgtm

**Pull request merged**